plugins/module_utils/nsxt_base_resource.py

```diff
@@ -549,19 +565,47 @@ def _get_base_arg_spec_of_resource(self):
     def _extract_nsx_resource_params(self, resource_params):
         # extract the params belonging to this resource only.
         filtered_params = {}
-
-        def filter_with_spec(spec):
+        def filter_with_spec(spec, filtered_params, resource_params):
             for key in spec.keys():
                 if (key in resource_params and
                         resource_params[key] is not None):
-                    filtered_params[key] = resource_params[key]
+                    v = resource_params[key]
+                    current_spec = spec[key]
+                    current_spec_type = current_spec.get("type")
+                    if current_spec_type == 'dict':
+                        params = filter_with_spec(
+                            current_spec, {}, v)
+                        if params:
+                            filtered_params[key] = params
+                    elif current_spec_type == 'list' and v:
+                        if isinstance(v[0], dict):
```

Comment:
This might deserve a comment for improving readability.

for instance - assuming that if the first element of the list is a dict, then all the other element will be dicts as well

Reply:
Sure, they will be dict though because spec looks like this in such case:
```
                next_hops=dict(
                    required=True,
                    type='list', <--------- current_spec_type
                    elements='dict', <--- elements can be all str, dict, or list. This defaults to str
                    options=dict(
                        admin_distance=dict(
                            type='int',
                            default=1
                        ),
                        ip_address=dict(
                            type='str'
                        ),
                        scope=dict(
                            type='list',
                            elements='str'
                        )
                    )
                ),
```

I have updated the condition so that the comment is not needed.